zerver/tests/test_user_groups.py

```diff
@@ -1125,7 +1206,79 @@ def check_update_user_group(
         othello.save()
         check_update_user_group("support", "Support team", "othello")
 
-    def test_user_group_edit_policy_for_updating_members(self) -> None:
+    def test_group_level_setting_for_updating_user_groups(self) -> None:
+        othello = self.example_user("othello")
+        iago = self.example_user("iago")
+        user_group = check_add_user_group(
+            get_realm("zulip"), "support", [othello, iago], acting_user=othello
+        )
+
+        def check_update_user_group(
+            new_name: str,
+            new_description: str,
+            acting_user: str,
+            error_msg: Optional[str] = None,
+        ) -> None:
+            self.login(acting_user)
+            params = {
+                "name": new_name,
+                "description": new_description,
+            }
+            # Ensure that this update request is not a no-op.
+            self.assertNotEqual(user_group.name, new_name)
+            self.assertNotEqual(user_group.description, new_description)
+
+            result = self.client_patch(f"/json/user_groups/{user_group.id}", info=params)
+            if error_msg is None:
+                self.assert_json_success(result)
+                user_group.refresh_from_db()
+                self.assertEqual(user_group.name, new_name)
+                self.assertEqual(user_group.description, new_description)
+            else:
+                self.assert_json_error(result, error_msg)
+
+        realm = othello.realm
+
+        do_set_realm_property(
+            realm,
+            "user_group_edit_policy",
+            Realm.POLICY_NOBODY,
+            acting_user=None,
+        )
+
+        # Default value of can_manage_group is single user group corresponding to othello.
+        check_update_user_group("help", "Troubleshooting team", "iago", "Insufficient permission")
+        check_update_user_group("help", "Troubleshooting team", "othello")
+
+        system_group_dict = get_role_based_system_groups_dict(realm)
+
+        user_group.can_manage_group = system_group_dict[SystemGroups.OWNERS]
+        user_group.save()
+        check_update_user_group("support", "Support team", "iago", "Insufficient permission")
+        check_update_user_group("support", "Support team", "desdemona")
+
+        user_group.can_manage_group = system_group_dict[SystemGroups.ADMINISTRATORS]
+        user_group.save()
+        check_update_user_group("help", "Troubleshooting team", "shiva", "Insufficient permission")
+        check_update_user_group("help", "Troubleshooting team", "iago")
+
+        user_group.can_manage_group = system_group_dict[SystemGroups.MODERATORS]
```

Comment:
These tests were added originally in the PR. I think we can remove these since it essentially just tests `is_user_in_group` function, but am not sure so have kept this for now.